### DISCUSSION TITLE
Easiest way to determine driver position at any point in time during the race

### DISCUSSION THREAD

**Discussion by marcdemar** [theOehrly/Fast-F1#284] [Q&A] [answered]
I'd like to be able to rank drivers by their on-track position (at any given time). Is there some way of doing that using telemetry/position data? I was thinking that perhaps Distance and RelativeDistance could be used for that, but that may not give a totally accurate result.  Any suggestions for how this could be done? Thanks!

**Comment by theOehrly** [comment, marked as answer]
Do you just want to know the correct order? If so, the next question is where should lapped cars be in that order? Do you want to know distances between cars as well?

**Reply by marcdemar** [reply]
I mean the correct race order. If car x is 15th at some point, I'd like to know this (regardless of whether the car has been lapped or not). As far as distances and timing is concerned, yes I'd like to know this as well (gaps to race leader, intervals). What I am trying to achieve is doing a replay and have the correct race order, gaps and intervals displayed at any time during the race.

**Reply by theOehrly** [reply]
You'll want to use the function `fastf1.api.timing_data`. It returns two values. The second one (called "stream data" in the docs) gives you the data that you are looking for. See https://theoehrly.github.io/Fast-F1/api.html#fastf1.api.timing_data

This is basically the raw data that is returned by the API. You will need to process the data further yourself. And you will need to do interpolation to get values at arbitrary timestamps.

**Reply by marcdemar** [reply]
Awesome, thanks. I will look into that.
